app/src/common/shared/org/mozilla/vrbrowser/browser/engine/SessionStore.java

```diff
@@ -89,15 +112,29 @@ public void onTrackingProtectionLevelUpdated(int level) {
                 });
             }
         });
-    }
 
-    public void initializeServices() {
-        mServices = ((VRBrowserApplication)mContext.getApplicationContext()).getServices();
-    }
+        mWebExtensionRuntime = new GeckoWebExtensionRuntime(mContext, mRuntime);
+        mWebExtensionRuntime.registerWebExtensionDelegate(this);
+
+        mServices = ((VRBrowserApplication)context.getApplicationContext()).getServices();
 
-    public void initializeStores(Context context) {
         mBookmarksStore = new BookmarksStore(context);
         mHistoryStore = new HistoryStore(context);
+
+        // Web Extensions initialization
+        YoutubeExtension.install(mWebExtensionRuntime);
+        VimeoExtension.install(mWebExtensionRuntime);
```

Comment:
This pattern doesn't scale. If the WebExtenstionRuntime API changes we have to update every class. I especially dislike it because the only difference between the two classes is the meta data.

Reply:
We can add a builtin extensions manager that initializes all of them based on metadata as we had before. I don't expect that we have many builtin extensions though mot of them are going to come from AC or the Addon Manager in the future, so I'm not sure if it'll be that useful.

Reply:
Or just have a single class that takes the meta data in the constructor. I guess it doesn't matter.

Reply:
Fixed, I've added a BuiltinExtension class that handles the initialization for our builtin extensions.